### DISCUSSION TITLE
Run `llama.cpp` with `tea` – without the installation pain!

### DISCUSSION THREAD

**Discussion by mfts** [ggml-org/llama.cpp#487] [Show and tell]
First of all, on behalf of open-source developers and users, thank you so much for porting LLaMA to C++ ❤️

## Running open-source made easy 

At tea[^1], we love open-source, so we packaged up `llama.cpp` and download the **7B model** via torrents. 
All you need to get started is

1. Install tea: `sh <(curl https://tea.xyz)`
2. Run `llama.cpp -p "Getting paid to write open source can be accomplished in 3 simple steps:"`

### How it works
By running the package command `llama.cpp -p "..."`, tea automagically downloads all required dependencies and runs `llama.cpp` for you. All dependencies get downloaded into an isolated `~/.tea` directory, which doesn't mess with your system installations.

If you are curious, you can take a look at the package file for llama.cpp [here](https://github.com/teaxyz/pantry/tree/main/projects/github.com/ggerganov/llama.cpp).

[^1]: [`tea`](https://github.com/teaxyz/cli) is the next-generation cross-platform package manager from the creator of Homebrew.

**Comment by Waloumi** [comment]
where do i run this? cmd? please...

**Comment by SolsticeProjekt** [comment]
Hi. Found you. Got this:

04/19 23:33:07 [NOTICE] Downloading 1 item(s)

04/19 23:33:07 [ERROR] Exception caught while loading DHT routing table from /home/bla/.cache/aria2/dht.dat
Exception: [DHTRoutingTableDeserializer.cc:83] errorCode=1 Failed to load DHT routing table from /home/bla/.cache/aria2/dht.dat

04/19 23:33:07 [NOTICE] IPv4 DHT: listening on UDP port 6997

04/19 23:33:07 [NOTICE] IPv4 BitTorrent: listening on TCP port 6957

04/19 23:33:07 [NOTICE] IPv6 BitTorrent: listening on TCP port 6957
[#7fd220 0B/0B CN:0 SD:0 DL:0B]
[#7fd220 0B/0B CN:0 SD:0 DL:0B]
[#7fd220 0B/0B CN:0 SD:0 DL:0B]

I don't know if it actually does something, though.

edit: CTRL+C, running the command again ... and the error message vanished.
It's still not actually doing anything, though, but i'll give it some time.

Any infos would be greatly appreciated.

**Reply by ampersanda** [reply]
I think there's a problem with the magnet url of the llama.cpp torrent
